### PR TITLE
feat(shared-registration): [skp-211] create registration component

### PR DESCRIPTION
**Pull request by doncevas**

### CODE REVIEW

File: libs/shared/registration/src/lib/shared-registration.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import Person from '@material-ui/icons/Person';
```
Comment (MariuzM) [inline libs/shared/registration/src/lib/shared-registration.tsx:3]: All Icons can be grouped

File: libs/shared/registration/src/lib/shared-registration.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  .card {
+    align-self: center;
+    max-width: 400px;
+    -webkit-box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.75);
+    -moz-box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.75);
+    box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.75);
+    border-radius: 5px;
+    padding: 20px;
+  }
+  .MuiBox-root {
```
Comment (MariuzM) [inline libs/shared/registration/src/lib/shared-registration.scss:17]: Not sure if using this `hacky` way of doing css for Material UI component is good idea.

File: libs/shared/registration/src/lib/shared-registration.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+export const SharedRegistration = (props: SharedRegistrationProps) => {
+  const { handleSubmit, register, errors, watch } = useForm();
+
+  const submit = (data: {
+    email: string;
+    password: string;
+    repeatPassword: string;
+    phone?: string;
```
Comment (MariuzM) [inline libs/shared/registration/src/lib/shared-registration.tsx:33]: Types could be defined at top level so the code looks cleaner

File: libs/shared/registration/src/lib/shared-registration.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+}
+
+interface RegistrationFormData{
+  email: string;
+  password: string;
+  repeatPassword: string;
+  phone?: string;
+}
+
+interface SharedRegistrationProps {
```
Comment (Gardemarin) [inline libs/shared/registration/src/lib/shared-registration.tsx:29]: add export on front of it, so you could import it inside shared-registration.stories.tsx: 
Screenshot 2020-07-27 at 19 10 30

File: libs/shared/registration/src/lib/shared-registration.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+              name="email"
+              InputProps={{
+                startAdornment: (
+                  <InputAdornment position="start">
+                    <Person />
+                  </InputAdornment>
+                ),
+              }}
+              fullWidth
+              type="text"
```
Comment (Gardemarin) [inline libs/shared/registration/src/lib/shared-registration.tsx:61]: change to type="email"

File: libs/shared/registration/src/lib/shared-registration.tsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+                pattern: {
+                  value: /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+                  message: 'Enter a valid e-mail address',
+                },
+              })}
+              error={!!errors.email}
+              label={!errors.email ? 'Email *' : 'Error'}
+              helperText={!errors.email ? '' : errors.email.message}
+            />
+            {console.log(errors.email)}
```
Comment (Gardemarin) [inline libs/shared/registration/src/lib/shared-registration.tsx:74]: Why you use console.log() here? Remove console.log().

File: libs/shared/registration/src/lib/shared-registration.tsx [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+                validate: (value) =>
+                  value === watch('password') || "Passwords don't match.",
+              })}
+              error={!!errors.repeatPassword}
+              label={!errors.repeatPassword ? 'Repeat Password *' : 'Error'}
+              helperText={
+                !errors.repeatPassword ? '' : errors.repeatPassword.message
+              }
+            />
+            {console.log(errors.repeatPassword)}
```
Comment (Gardemarin) [inline libs/shared/registration/src/lib/shared-registration.tsx:126]: Remove console.log()

File: libs/shared/registration/src/lib/shared-registration.tsx [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+              placeholder="Phone"
+              fullWidth
+              InputProps={{
+                startAdornment: (
+                  <InputAdornment position="start">
+                    <PhoneIcon />
+                  </InputAdornment>
+                ),
+              }}
+              type="string"
```
Comment (Gardemarin) [inline libs/shared/registration/src/lib/shared-registration.tsx:142]: Change type to type="tel"

File: libs/shared/registration/src/lib/shared-registration.tsx [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+  Box,
+  Grid,
+  TextField,
+  Typography,
+  InputAdornment,
+  Button,
+} from '@material-ui/core';
+import Person from '@material-ui/icons/Person';
+import LockIcon from '@material-ui/icons/Lock';
+import PhoneIcon from '@material-ui/icons/Phone';
```
Comment (Gardemarin) [inline libs/shared/registration/src/lib/shared-registration.tsx:13]: Import all icons in one import: 

import {
  Person,
  Lock as LockIcon,
  Phone as PhoneIcon,
} from '@material-ui/icons';

File: libs/shared/registration/src/lib/shared-registration.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+export const primary = () => {
+  const onSubmit = action('onSubmit');
+
+  
+  const props: SharedRegistrationProps = {
+    onSubmit
+  };
+  
+
+  return <SharedRegistration onSubmit = {props.onSubmit}  />;
```
Comment (Gardemarin) [inline libs/shared/registration/src/lib/shared-registration.stories.tsx:23]: you can make it simpler: onSubmit = {props.onSubmit} change to  {...props}

**Pull request merged**